src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Utility.php

```diff
@@ -89,6 +89,11 @@ protected function loadQuoteByCustomer($validate = null)
                      ->loadByCustomer($this->getApiUser()->getUserId());
 
         $quote = $quote->getId() ? $quote : $this->loadInactiveCustomerQuote();
+
+        if (!$quote->getId()) {
+            $quote = $this->createNewCustomerQuote($this->getApiUser()->getUserId());
```

Comment:
I will have to double check this logic, most likely loading an inactive quote is detrimental in this case. This is also a utility class, so I am not sure which endpoints use this call yet.

Reply:
i have tested this logic and its works in my case. If you have a better place / logic please inform me.

Reply:
in a bit more detail:
```
$quote = $quote->getId() ? $quote : $this->loadInactiveCustomerQuote();
```
The above code was created in response to our issue when a customer makes a purchase and the quote is no longer active. The subsequent calls to -> 
v2/carts/me
v2/carts/:guestCartId/customer/
/v2/carts/me/checkoutUrl
will need to return an active quote if it exists or return an inactive quote if no active quote is present.

What your logic adds is this -> if no active or inactive quote is found -> create a new cart for this customer. So the logic will only work when there are no quotes at all for the customer.

So my question, is this the proper logic? Do we still need to load an inactive quote or should we just create a new active quote every time instead? If an inactive quote is returned, do the pipelines handle that and try not to add products to an inactive cart, and afterwards make a call to make a new cart?

Reply:
also, I think a postman test will need to be created to replicate the error and see if it passes after the fix. The postman test collection is on the develop branch.